lib/luadata.c

```diff
@@ -85,6 +92,9 @@ static int luadata_setstring(lua_State *L)
 	const char *str = luaL_checklstring(L, 3, &length);
 	luadata_t *data = luadata_checkdata(L, &offset, (lua_Integer)length);
 
+	if (!data->editable)
+		luaL_error(L, "data is not editable");
```

Comment:
you can move this to an inline, right? see luasocket for this sort of checker example..